fhd_core/source_modeling/source_dft_multi.pro

```diff
@@ -50,78 +52,43 @@ FOR a_i=0L,n_alpha-1 DO BEGIN
 ENDFOR
 
 IF keyword_set(gaussian_source_models) then begin
-  if tag_exist(source_array, 'shape') then begin
-    gaussian_ra=make_array(n_elements(source_array),/double)
-    gaussian_dec=make_array(n_elements(source_array),/double)
-    gaussian_rot=make_array(n_elements(source_array),/double)
-    for source_ind=0,n_elements(source_array)-1 do begin
-      if tag_exist(source_array[source_ind], 'shape') then begin
+    if tag_exist(source_array, 'shape') then begin
+        gauss_test = ceil(0>(source_array.shape.x + source_array.shape.y)<1)
+        gauss_inds = where(gauss_test GT 0,n_gauss)
+    endif else n_gauss=0
+
+    if n_gauss GT 0 then begin
         ;Convert from FWHM in arcsec to stddev in deg
-        gaussian_ra[source_ind]=source_array[source_ind].shape.x/(7200*sqrt(2*alog(2)))
-        gaussian_dec[source_ind]=source_array[source_ind].shape.y/(7200*sqrt(2*alog(2)))
+        gaussian_ra=double(source_array[gauss_inds].shape.x)/(7200*sqrt(2*alog(2)))
+        gaussian_dec=double(source_array[gauss_inds].shape.y)/(7200*sqrt(2*alog(2)))
         ;Convert from deg to rad
-        gaussian_rot[source_ind]=source_array[source_ind].shape.angle*!Pi/180.
-      endif else begin
-        gaussian_ra[source_ind]=0
-        gaussian_dec[source_ind]=0
-        gaussian_rot[source_ind]=0
-      endelse
-    endfor
-    inds_gauss=where(gaussian_ra,n_gauss_params,complement=inds_point)
-    if n_gauss_params eq 0 then begin
-      print, 'Catalog does not contain Gaussian shape parameters. Unsetting keyword gaussian_source_models.'
-      undefine, gaussian_source_models
-    endif else begin
+        gaussian_rot=double(source_array[gauss_inds].shape.angle)*!DPi/180.
```

Comment:
Here and throughout this file, please use `!Radeg` instead of `!DPi/180`

Reply:
Unfortunately, `!Radeg=57.295776` is floating precision, so I would rather stay with `!Dpi/180=57.295779513082323`.